File: frontend/src/types/backtest.ts
export interface BacktestTaskBase {
    strategy_id: string;
    name: string;
    start_date: string;  // YYYY-MM-DD
    end_date: string;    // YYYY-MM-DD
    initial_capital: number;
    config?: Record<string, any>;
}

export interface BacktestTaskCreatePayload extends BacktestTaskBase {}

export interface BacktestMetrics {
    total_returns?: number;
    max_drawdown?: number;
    total_trades?: number;
}

export interface BacktestEquityPoint {
    date: string;
    equity: number;
}

export interface BacktestResult {
    equity_curve?: BacktestEquityPoint[];
    metrics?: BacktestMetrics;
}

export interface BacktestTask extends BacktestTaskBase {
    id: string;
    status: 'pending' | 'running' | 'completed' | 'failed';
    progress: number;
    error_message?: string;
    result?: BacktestResult;
    created_at?: string;
}

export interface BacktestTrade {
    id: string;
    backtest_id: string;
    symbol: string;
    side: 'buy' | 'sell';
    quantity: number;
    price: number;
    commission?: number;
    trade_date: string;
}

export interface BacktestTaskWithTrades extends BacktestTask {
    trades: BacktestTrade[];
}

export interface BacktestListResponse {
    total: number;
    items: BacktestTask[];
}

File: backend/schemas/backtest.py
import uuid
from typing import Dict, Any, Optional, List
from datetime import date
from pydantic import BaseModel, Field

# ========= Backtest Trade =========

class BacktestTradeBase(BaseModel):
    order_book_id: Optional[str] = None
    symbol: str
    side: str
    quantity: int
    price: float
    commission: float = 0.0
    trade_date: date

class BacktestTradeResponse(BacktestTradeBase):
    id: uuid.UUID
    backtest_id: uuid.UUID
    
    model_config = {"from_attributes": True}


# ========= Backtest Task =========

class BacktestTaskBase(BaseModel):
    strategy_id: uuid.UUID
    name: str = Field(..., max_length=100)
    start_date: date
    end_date: date
    initial_capital: float = Field(..., gt=0)
    config: Dict[str, Any] = Field(default_factory=dict)

class BacktestTaskCreate(BacktestTaskBase):
    pass

class BacktestTaskUpdate(BaseModel):
    status: Optional[str] = None
    result: Optional[Dict[str, Any]] = None
    error_message: Optional[str] = None
    progress: Optional[int] = None

class BacktestTaskResponse(BacktestTaskBase):
    id: uuid.UUID
    status: str
    result: Optional[Dict[str, Any]] = None
    error_message: Optional[str] = None
    progress: int
    created_at: Optional[Any] = None

    model_config = {"from_attributes": True}

class BacktestTaskWithTradesResponse(BacktestTaskResponse):
    trades: List[BacktestTradeResponse] = []
